### PR TITLE
feat: create initial dashboard page

### PR DESCRIPTION
**Pull request by karrui**
### TL;DR
This PR adds a new `/dashboard` page to go to a specific site

### What changed?
1. **Site Listing**:
    - Introduced a new `SiteList` component that lists all sites a user has access to.
    - Added new database query in `site.router.ts` to fetch site list.
2. **Report Issue Button**:
    - Added a button in the AppNavbar to allow users to report issues via the Isomer issue tracker.
3. **Component Updates**:
    - Updated `AppNavbar` to include new functionality and style tweaks.
    - Changed the logo source in `AppNavbar.tsx`.
    - Updated `dashboard.tsx` with new site listing component.
    - Refactored various storybook files to include new handlers and remove unnecessary layouts.

### How to test?
New stories

---

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/308?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#309** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/309?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#308** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/308?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#302** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/302?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#301** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/301?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/308?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#309** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/309?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#308** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/308?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#302** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/302?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#301** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/301?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/studio/src/pages/dashboard.tsx [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
+        <Text as="h3" size="lg" textStyle="h3">
+          My sites
+        </Text>
+        <Text textStyle="body-2">
+          Don't see a site here?{" "}
+          <Link variant="inline" href="mailto:support@isomer.gov.sg">
+            Let us know
+          </Link>
+          .
+        </Text>
```
Comment (dcshzj) [inline apps/studio/src/pages/dashboard.tsx:21]: question: not sure if design changed anything here but this message seems to only appear if the user has access to at least one site, and it's not shown if the user sees 0 sites.

Should the site list query be moved here instead, and SiteList takes the list of sites as a prop?

Comment (karrui) [reply apps/studio/src/pages/dashboard.tsx:21]: Dashboard design not ready yet, creating this PR as a placeholder for routing first haha

Comment (dcshzj) [reply apps/studio/src/pages/dashboard.tsx:21]: Ah okay cos on Figma I saw as ready for dev, will approve first

**Review by dcshzj — APPROVED** [review]

File: apps/studio/src/pages/dashboard.tsx (hunk 1 above)
Comment (karrui) [reply apps/studio/src/pages/dashboard.tsx:21]: Will do in later PRs!

File: apps/studio/public/assets/isomer-logo.svg
Comment (seaerchin) [inline apps/studio/public/assets/isomer-logo.svg:1]: why black colour

File: apps/studio/src/features/dashboard/SiteList.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          <Skeleton>
+            <CardHeader>Loading...</CardHeader>
+          </Skeleton>
+        </Card>
+      ))}
+    </SimpleGrid>
+  )
+}
+
+export const SiteList = withSuspense(SuspendableSiteList, <SiteListSkeleton />)
```
Comment (seaerchin) [inline apps/studio/src/features/dashboard/SiteList.tsx:40]: very cool

**Review by seaerchin — COMMENTED** [review]
looks good, things that i thought was interesting was the use of `Suspense` here

**Comment by karrui** [comment]
### Merge activity

* **Jul 18, 11:17 PM EDT**: @karrui started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/308).
